### ISSUE TITLE
right/left alignment of equations doesn't work in `cases` function

### ISSUE THREAD

**Issue by rinmyo** [typst/typst#1478]
### Description

#936 

### Reproduction URL

https://typst.app/project/r7rsce5HBrrfp4EmN81gML

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by user202729** [comment]
It doesn't make sense to alternate right/left alignment in `cases` right? It's intended that you write something like `cases(1+2&"if" 3>4, 4+5+6&"otherwise")` and it makes more sense to left align the first column.

**Comment by uwni** [comment]
just like the instance posted in the Reproduction URL.
when I want to align saperately the variables and parameters of a equation system. it should be expected to act like https://mirrors.ctan.org/macros/generic/systeme/systeme_doc_fr.pdf
this, in my opinion

**Comment by user202729** [comment]
that's semantically not a `cases`, and it's an abuse to use `cases` anyway. What you want is the equivalent of LaTeX's `aligned` environment, which as far as I can see Typst has no direct equivalent alternative, but you can make one with a nested `box`:

```
$a+ lr( \{ #box[$ b+f &= c \ d &= e $]  ) $
```

**Comment by laurmaedje** [comment]
Cases intentionally does not alternate between left and right.

**Comment by Andrew15-5** [comment]
I think this is a perfectly valid use case for `cases` that I unfortunately have to hack myself through:

```typ
#let cur = `current`
#let lvl = `level`

$ lr(\{#block($
  cur + lvl * 2, & lvl < 0.5; \
  cur + (1 - cur) * (lvl - 0.5) * 2, & lvl gt.eq.slant 0.5.
$)) $
```
![image](https://github.com/user-attachments/assets/c4590961-f672-4075-a264-19c74b28b30a)

Thanks to @user202729 for the base code. `box` messes up the height of the inner container and the length of the curly brace. `block` fixes it. I also think that making inner `math.equation` block-level looks more readable, but you can also use inline (which feels more cramped).

P.S. In Russian books, I think it's a standard to use comma for separating the value and the condition, and then separating lines with semicolon. Then ending the last line with a punctuation that is appropriate in the given context (usually a period). So if I _had to_ use `cases`, I would've escaped a bunch of stuff, which is never clearer than no escaping.

**Comment by pauladam94** [comment]
> Cases intentionally does not alternate between left and right.

I don't understand why is that ? I found my self doing this like @Andrew15-5 did in https://github.com/typst/typst/issues/1478#issuecomment-2400899653.

Like in any other equation, why would cases be fixed to left alignment ?

Let's say cases respect the classic behavior of equation. If someone really wants this behavior (always left aligned) and use control point, it can just do a classic "trick" to always use `&&` instead of `&` to be sure to have left alignment every time.

I don't see any advantage to change the behavior of the alignment in equation only inside a specific function like `cases`.

I first thought this was a bug, I looked extensively inside the docs before looking at the issue in github.

This part of the docs from [cases docs](https://typst.app/docs/reference/math/cases/) is clearly misleading. It should at least be stated in the documentation that cases does not respect the same behavior as the other equation. (Something we don't really want in the docs because it seems weird to have different behavior in random places in equation).
<img width="860" height="222" alt="Image" src="https://github.com/user-attachments/assets/cd6ed378-9dcf-41e7-b421-2d053655e478" />
